staging/src/k8s.io/client-go/tools/cache/listers.go

```diff
@@ -32,6 +32,10 @@ type AppendFunc func(interface{})
 
 // ListAll lists items in the store matching the given selector, calling appendFn on each one.
 func ListAll(store Store, selector labels.Selector, appendFn AppendFunc) error {
+	if selector == labels.Nothing() {
```

Comment:
Comparing an interface instance to a concrete value to determine if it is the special-case "match nothing" is a bit odd. I'm not sure whether it is better to add a new method to the labels.Selector interface so implementations can identify this case, or add a helper `func MatchesNothing(selector labels.Selector) bool` function to the labels package so we can keep the special comparison to the `nothingSelector` type local to the package.

cc @deads2k for an opinion

cc @munnerz who has spent some time thinking about optimizing / canonicalizing selectors (so something like `matchExpressions: [{"key":"foo","operator":"Exists"}, {"key":"foo","operator":"DoesNotExist"}]` would also correctly identify as matching nothing)

Reply:
External callers might also need to check whether the selector is labels.Nothing(). Would it be ok to add a new function to the interface, similar to Empty()?